src/core/MapControls.tsx

```diff
@@ -17,17 +17,24 @@ declare global {
 
 export const MapControls = React.forwardRef<MapControlsImpl, MapControls>((props = { enableDamping: true }, ref) => {
   const { camera, ...rest } = props
-  const invalidate = useThree(({ invalidate }) => invalidate)
-  const defaultCamera = useThree(({ camera }) => camera)
-  const gl = useThree(({ gl }) => gl)
+  const invalidate = useThree(({ invalidate }) => invalidate);
+  const defaultCamera = useThree(({ camera }) => camera);
+  const domElement = useThree(({ gl }) => gl.domElement);
+  
   const explCamera = camera || defaultCamera
-  const [controls] = React.useState(() => new MapControlsImpl(explCamera, gl.domElement))
+  const [controls] = React.useState(() => new MapControlsImpl(explCamera))
```

Comment:
While we are at it, there is another bug - controls camera is never updated in case it changes.
Meaning, if you do something like this (see `makeDefault`):
```ts
  const camera = useRef();

  return (
    <>
      <MapControls camera={camera.current} />
      <OrthographicCamera ref={camera} makeDefault {...CAMERA_PROPS} />
    </>
  );
```
change will be triggered by:
```ts
const defaultCamera = useThree(({ camera }) => camera);
```
however, `controls` will still use old `explCamera`.

Testing it locally, this seems to fix the problem:
```ts
  const explCamera = camera || defaultCamera
  const [controls] = React.useState(() => new MapControlsImpl(explCamera));
  controls.object = explCamera;
```

not sure if that is the best way, tho.

Reply:
We can swap to use useMemo which IMO would be better :)

Reply:
Changed `useState` to `useMemo`.